gensim/models/keyedvectors.py

```diff
@@ -1384,12 +1387,42 @@ def init_sims(self, replace=False):
             else:
                 self.vectors_norm = (self.vectors / sqrt((self.vectors ** 2).sum(-1))[..., newaxis]).astype(REAL)
 
+    def relative_cosine_similarity(self, wa, wb, topn=10):
+        """Compute the relative cosine similarity between two words given top-n similar words,
+        proposed by Artuur Leeuwenberg, Mihaela Vela, Jon Dehdari, Josef van Genabith
+        "A Minimally Supervised Approach for Synonym Extraction with Word Embeddings"
+        <https://ufal.mff.cuni.cz/pbml/105/art-leeuwenberg-et-al.pdf>.
+
+        To calculate relative cosine similarity between two words, equation (1) of the paper is used.
+        For WordNet synonyms, if rcs(topn=10) is greater than 0.10 then wa and wb are more similar than
+        any arbitrary word pairs.
+
+        Parameters
+        ----------
+        wa: str
+            word for which we have to look top-n similar word.
+        wb: str
+            word for which we evaluating relative cosine similarity with wa.
+        topn: int, optional
+            Number of top-n similar words to look with respect to wa.
+        Returns
+        -------
+        numpy.float64
+            relative cosine similarity between wa and wb.
+        """
+        sims = self.similar_by_word(wa, topn)
+        assert sims, "Cannot generate similar words"
```

Comment:
Thanks! Please make the message a little more concrete / verbose though.

As a user, if this message `Cannot generate similar words` jumped at me, I wouldn't know what that means.

How about `Failed code invariant: list of similar words must never be empty.` ?

This is of course assuming it truly is an invariant (can never happen, as opposed to "shouldn't happen on most inputs").

Reply:
Thanks. I have made the changes.